contrib/babelfishpg_common/src/varbinary.c

```diff
@@ -190,6 +191,13 @@ varbinaryin(PG_FUNCTION_ARGS)
 		PG_RETURN_BYTEA_P(result);
 	}
 
+	if (!(IsTdsClientHook && (*IsTdsClientHook)()) && 
```

Comment:
we can have direct hook between common and TDS/TSQL using rendevouz variables. No need to go via engine. Also, ideal way to do it is

Find out what logic is specific to TDS vs PSQL
Add a function in TDS to do TDS specific work
Add a rendevouz variable between common and TDS to access the above function
in common
if (tds_hook && tds_hook())
else
   PSQL logic

However, if we have time constraints, just add a hook via rendevouz variable

Reply:
Oh cool, I didn't know about the existence of rendezvous variables. I'll update the implementation to use them.

Reply:
Updated to check `psql_logical_babelfish_db_name` GUC instead